src/Compilers/CSharp/Portable/Binder/Binder_Deconstruct.cs

```diff
@@ -705,6 +704,17 @@ private static void FlattenDeconstructVariables(ArrayBuilder<DeconstructionVaria
                         }
                         return new DeconstructionVariable(builder, node);
                     }
+                case SyntaxKind.IdentifierName:
+                    {
+                        var identifier = (IdentifierNameSyntax)node;
```

Comment:
You should attempt to bind it to an existing variable before treating it as a discard.